frontend/src/general/__tests__/SnackbarMessage.test.jsx

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+import Enzyme from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+import {
+  afterEach,
+  beforeEach,
+  describe,
+  expect,
+  it,
+  jest
+} from "@jest/globals";
+import { Snackbar, SnackbarContent, Box } from "@material-ui/core";
+import { CheckCircle, Error } from "@material-ui/icons";
+import SnackbarMessage from "../SnackbarMessage";
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe("<SnackbarMessage /> Class Component", () => {
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = Enzyme.shallow(<SnackbarMessage />);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    wrapper.unmount();
+  });
+
+  describe("render() function", () => {
+    describe("Returns a component that", () => {
+      it("Contains 1 <Snackbar/> component with the expected props ", () => {
+        const snackbarComponent = wrapper.find(Snackbar);
+        expect(snackbarComponent).toHaveLength(1);
+
+        const snackbarComponentProps = snackbarComponent.props();
+        expect(snackbarComponentProps.contentprops).toEqual({
+          "aria-describedby": "message-id"
+        });
+        expect(snackbarComponentProps.anchorOrigin).toEqual({
+          vertical: "top",
+          horizontal: "right"
+        });
+        expect(snackbarComponentProps.open).toBe(wrapper.state().open);
+        expect(snackbarComponentProps.autoHideDuration).toEqual(3000);
+        expect(snackbarComponentProps.onClose).toBe(
+          wrapper.instance().handleClose
+        );
+
+        expect(wrapper.find(SnackbarContent)).toHaveLength(1);
+      });
+
+      describe("Contains 1 <SnackbarContent/> component where", () => {
+        const statusColors = {
+          green: "#4caf50",
+          red: "#f44336"
+        };
+        describe("If status is success,", () => {
+          describe("the snackbar has a green background and the correct props", () => {
+            it("and contains one <CheckCircle/> component", () => {
+              wrapper.setState({ status: "success" });
+              expect(wrapper.find(SnackbarContent).props().style).toEqual({
+                backgroundColor: statusColors.green
+              });
+            });
+          });
```

Comment:
I think you got mixed up here - your `it` mentions the CheckCircle component but doesn't actually check it's there.
Also you have two nested describes for a single test. The purpose of describe is to separate tests into groups, so having 2 of them surrounding one test is kind of missing the point.

You're almost there though! Try organizing the tests like this

```
describe("<SnackbarContent/> component")
    describe("If status is success")
        it("has a green background")
        it("contains a message prop with a <CheckCircle/> component")
    describe("If status is not success")
        it("Has a red background")
        it("Contains a message prop with a <Error/> component")
```

Reply:
Oh okay, I think I misunderstood your previous comments. Thanks for clarifying! I'll update accordingly.